src/store/movies-store/getMovies.thunk.ts

```diff
@@ -0,0 +1,17 @@
+import {Pagination} from "../../api/shared/models/pagination.model";
+import {moviesSlice} from "./movies.slice";
+import {moviesService} from "../../api/movies-api/movies.service";
+
+export const getMoviesThunk = (pagination: Pagination) => {
+  return async (dispatch: any) => {
+    dispatch(moviesSlice.actions.getMovies(pagination))
+
+    try {
+      const { data } = await moviesService.getMovies(pagination);
```

Comment:
I would change `getMovies` in slice or service. Now, it easy to be confused in case of the first reading. I got an impression that you are firing the same action twice.

Reply:
Don't think so